### PR TITLE
[GH-1401] Add `retry` Attribute to TerraExecutorDetails Type and Instances Thereof

### PR DESCRIPTION
**Pull request by ehigham**
RR: https://broadinstitute.atlassian.net/browse/GH-1401
Verified that this change affects all existing tables created from that type.
When workflows are requested, only the non-retried workflows are returned.
Note that unconsumed workflows are still visible to downstream. Note sure what the correct behaviour should be here.

### CODE REVIEW

**Review by rexwangcc — APPROVED** [review]
LGTM! (updated the [chart](https://lucid.app/lucidchart/invitations/accept/inv_39e2a1d7-c5c8-4cfc-86a4-7ea92c24a452) accordingly)

File: api/src/wfl/executor.clj [hunk 1]
```diff
@@ -332,7 +332,10 @@
   [tx {:keys [details] :as executor}]
   (when-not (postgres/table-exists? tx details)
     (throw (ex-info "Missing executor details table" {:table details})))
-  (let [query "SELECT * FROM %s WHERE workflow IS NOT NULL ORDER BY id ASC"]
+  (let [query "SELECT * FROM %s
+               WHERE workflow IS NOT NULL
+               AND   retry    IS NULL
```
Comment (okotsopoulos) [inline api/src/wfl/executor.clj:337]: Can you explain why we want to exclude records with a populated `retry` from these fetches?

**Comment by ehigham** [comment]
> LGTM! (updated the [chart](https://lucid.app/lucidchart/invitations/accept/inv_39e2a1d7-c5c8-4cfc-86a4-7ea92c24a452) accordingly)

Thank you!

File: api/src/wfl/executor.clj (hunk 1 above)
Comment (ehigham) [reply api/src/wfl/executor.clj:380]: Added a long comment

**Pull request merged**